frontend/public/components/utils/vertnav.jsx

```diff
@@ -41,6 +42,11 @@ export const navFactory = {
     name: 'YAML',
     component: component,
   }),
+  viewYaml: (component = viewYamlComponent) => ({
```

Comment:
You can just pass `viewYaml` component to the existing editYaml factory. It takes in the component as a parameter